### PR TITLE
Upgrade to Next.js 13.4

### PR DESCRIPTION
**Pull request by taysea**
[Deploy Preview](https://deploy-preview-3397--keen-mayer-a86c8b.netlify.app/)

#### What does this PR do?

Upgraded to Next.js 13 and other related dependency upgrades (mdx packages, etc.).

Fixed lint errors, enabled react strict to enforce best practices earlier.

Notes:
- In latest MDX, `.isMDXComponent` is deprecated, so now there is a single return which then required some adjustments to "hub" pages (components index.js, templates index.js, ...), the Layout page, and _app to account for this
- In latest MDX (v2), comment formatting isn't support in the same way as v1 so I made adjustments to adhere to the latest support
- Fixed the return in ThemeMode to fix react hydration errors
- Remove babel config to leverage Next.js compiler: https://nextjs.org/docs/architecture/nextjs-compiler

To do:
- [ ] The rule definitions in eslint-plugin-grommet were not able to found and were causing some lint errors in next.config.js
- [ ] Fix linting errors in ModalTaskFlow (temporarily disabled)

#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
I'm seeing a double header on the learn page
Screen Shot 2023-06-12 at 9 44 20 AM

Also it looks like the what's new page lost its padding
Screen Shot 2023-06-12 at 9 45 42 AM

**Comment by taysea** [comment]
> I'm seeing a double header on the learn page Screen Shot 2023-06-12 at 9 44 20 AM
> 
> Also it looks like the what's new page lost its padding Screen Shot 2023-06-12 at 9 45 42 AM

Good catch, thanks I'll audit the rest of the "non-MDX" pages to make sure they come through properly.

File: .eslintrc.js [hunk 1]
```diff
@@ -12,7 +12,8 @@ module.exports = {
     'airbnb',
     'plugin:jsx-a11y/recommended',
     'prettier',
-    'plugin:grommet/recommended',
+    // TO DO resolve why definitions weren't found
```
Comment (jcfilben) [inline .eslintrc.js:15]: Should we create a separate issue to look into this?

**Review by jcfilben — APPROVED** [review]
Looks good!

File: .eslintrc.js (hunk 1 above)
Comment (taysea) [reply .eslintrc.js:15]: Yes: https://github.com/grommet/hpe-design-system/issues/3403

**Pull request merged**